Gemfile

```diff
@@ -21,6 +21,8 @@ gem 'jbuilder', '~> 2.0'
 # bundle exec rake doc:rails generates the API under doc/api.
 gem 'sdoc', '~> 0.4.0', group: :doc
 gem 'pg'
+gem 'carrierwave'
```

Comment:
@arafatm It is the most flexible gem for handling the file uploads especially images, it has rich built in features which handles and processes the images as you wish and the most interesting part is that it isolates all of its stuffs by allowing you to `rails generate uploader Image` which will create `app/uploaders/image_uploader.rb` and all the files concerning the file uploads will go in there